File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/javadoc/uncomment/direct/Java/Go/s936984238_commented.go
package main

import (
	"fmt"
	"sort"
)

func main() {
	for {
		var n int
		fmt.Scan(&n)
		if n == 0 {
			break
		}
		scores := make([]int, n)
		for i := range scores {
			fmt.Scan(&scores[i])
		}
		sort.Ints(scores)
		sum := 0
		for i := 1; i < n-1; i++ {
			sum += scores[i]
		}
		fmt.Println(sum / (n-2))
	}
}

// End of Code

File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/javadoc/uncomment/direct/Java/Python/s936984238_commented.py
import sys

def main():
    data = list(map(int, sys.stdin.read().split()))
    idx = 0
    while True:
        n = data[idx]
        idx += 1
        if n == 0:
            break
        scores = data[idx:idx+n]
        idx += n
        scores.sort()
        s = sum(scores[1:-1])
        print(s // (n-2))

if __name__ == "__main__":
    main()
# End of Code
